tensorflow/core/lib/strings/numbers.cc

```diff
@@ -165,8 +112,8 @@ char* DoubleToBuffer(double value, char* buffer) {
     // larger than the precision we asked for.
     DCHECK(snprintf_result > 0 && snprintf_result < kFastToBufferSize);
 
-    full_precision_needed =
-        locale_independent_strtonum<double>(buffer, nullptr) != value;
+    auto parsed_value = double{};
```

Comment:
why this change?  (also, we could possibly use DoubleToStringConverter to implement all of DoubleToBuffer?)  (but see the question about android code size as well).

Reply:
The previous implementation of DoubleToBuffer function was using old locale_independent_strtonum<double> which is replaced by double_conversion::StringToDoubleConverter with slightly differen interface.

But it is a good point to use only one function to convert str to double everywhere here. Consider double_conversion::StringToDoubleConverter only as hidden implementation details of safe_strto{d,f}.

This is a good idea to use double-conversion for {Double,Float}ToBuffer functions. I can do it here or in another pull request.